src/codegen/cfg.rs

```diff
@@ -337,6 +338,7 @@ pub struct ControlFlowGraph {
     pub ty: pt::FunctionTy,
     pub selector: u32,
     current: usize,
+    pub temp_var_map: ArrayTempVars, //a new member in the cfg that track the temp vars of arrays
```

Comment:
This is not a good name. You don't need `_map` and it should really convey something about array lengths.

How about `array_length_temps`? @LucasSte what do you think?